### PR TITLE
Added SIGNUM function to V2 engine

### PR DESCRIPTION
**Pull request by matthewryanwells**
### Description
Added math function SIGNUM to V2 engine, updated documentation, and added unit/IT tests

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by GabeFernandez310 — APPROVED** [review]

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java [hunk 1]
```diff
@@ -1740,8 +1740,8 @@ public void round_null_missing() {
   public void sign_int_value(Integer value) {
     FunctionExpression sign = DSL.sign(DSL.literal(value));
     assertThat(
-        sign.valueOf(valueEnv()),
-        allOf(hasType(INTEGER), hasValue((int) Math.signum(value))));
+            sign.valueOf(valueEnv()),
```
Comment (GumpacG) [inline core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:1743]: Unnecessary changes?

**Review by GumpacG — APPROVED** [review]

**Comment by GumpacG** [comment]
Seems that there are doctests failing

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java (hunk 1 above)
Comment (matthewryanwells) [reply core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java]: fixed in `a68487c`

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java [hunk 2]
```diff
@@ -1804,6 +1804,84 @@ public void sign_missing_value() {
     assertTrue(sign.valueOf(valueEnv()).isMissing());
   }
 
+  /**
+   * Test signum with byte value.
+   */
+  @ParameterizedTest(name = "signum({0})")
+  @ValueSource(bytes = {2, -2})
```
Comment (Yury-Fridlyand) [inline core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:1811]: You don't have test for 0 argument

File: docs/user/dql/functions.rst [hunk 3]
```diff
@@ -760,9 +760,21 @@ SIGNUM
 Description
 >>>>>>>>>>>
 
-Specifications:
+Usage: Returns the sign of the argument as -1, 0, or 1, depending on whether the number is negative, zero, or positive
+
+Argument type: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE
+
+Return type: INTEGER
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:767]: synonym to `sign`

**Review by Yury-Fridlyand — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
Please, fix doctests

**Review by GabeFernandez310 — APPROVED** [review]

**Comment by matthewryanwells** [comment]
Doc tests have been fixed
